Orange/widgets/utils/__init__.py

```diff
@@ -1,13 +1,16 @@
 from functools import reduce
+from Orange.data.variable import TimeVariable
 
 
 def vartype(var):
     if var.is_discrete:
         return 1
-    elif var.is_continuous:
+    elif var.is_continuous and not isinstance(var, TimeVariable):
         return 2
     elif var.is_string:
         return 3
+    elif isinstance(var, TimeVariable):
+        return 4
```

Comment:
This abomination of a mechanism shall not retain its current form for much longer. But mending it now is out of scope in this PR.